### PR TITLE
Test to ensure proper compilation of abs function

### PR DESCRIPTION
**Pull request by fineg74**
The purpose of this test is to ensure proper compilation of abs function. No final results are checked.

### CODE REVIEW

**Comment by fineg74** [comment]
Complementary PR for https://github.com/intel/llvm/pull/6443

File: SYCL/ESIMD/regression/abs_fix_test.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// TODO online compiler check fails for esimd_emulator
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/abs_fix_test.cpp]: Is this TODO some copy-paste artefact? This test doesn't use online_compiler.

File: SYCL/ESIMD/regression/abs_fix_test.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#define SIMD 16
+#define THREAD_NUM 512
+
+template <typename DataT>
+using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
+template <typename DataT>
+using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
+
+template <typename T0, typename T1>
+int test_abs(cl::sycl::queue q, T1 test_value) {
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/abs_fix_test.cpp]:
```suggestion
int test_abs(sycl::queue q, T1 test_value) {
```

File: SYCL/ESIMD/regression/abs_fix_test.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  shared_allocator<T0> allocator0(q);
+  shared_vector<T0> B(THREAD_NUM * SIMD, allocator0);
+  for (int i = 0; i < A.size(); ++i) {
+    A[i] = test_value;
+  }
+
+  nd_range<1> Range((range<1>(THREAD_NUM)), (range<1>(SIMD)));
+  auto e = q.submit([&](handler &cgh) {
+    T1 *in_ptr = A.data();
+    T0 *const out_ptr = B.data();
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/abs_fix_test.cpp]: Shouldn't the 'const' specifier be used for 'in_ptr', not 'out_ptr' ?

File: SYCL/ESIMD/regression/abs_fix_test.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  return 0;
+}
+
+int main(int argc, char *argv[]) {
+  sycl::property_list properties{sycl::property::queue::enable_profiling()};
+  auto q = sycl::queue(properties);
+
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+  auto ctxt = q.get_context();
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/abs_fix_test.cpp]: This var is unused.
```suggestion
```

File: SYCL/ESIMD/regression/abs_fix_test.cpp [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+  int test_result = 0;
+  test_result |= test_abs<uint32_t, int32_t>(q, 0xFFFFFFFF);
```
Comment (v-klochkov) [inline SYCL/ESIMD/regression/abs_fix_test.cpp]: As far as I know, one of customer/reporter's concern was the case when only T1->T0 conversion is defined, this intrinsic is guaranteed to work. And having T0->T1 conversion is not required for abs to work.

Can you add such test where there is only one direction conversion to show that the intrinsic still works?

File: SYCL/ESIMD/regression/abs_fix_test.cpp (hunk 3 above)
Comment (fineg74) [reply SYCL/ESIMD/regression/abs_fix_test.cpp]: Technically here it is a constant pointer not a constant data. I believe it is not necessary here.

File: SYCL/ESIMD/regression/abs_fix_test.cpp (hunk 5 above)
Comment (fineg74) [reply SYCL/ESIMD/regression/abs_fix_test.cpp]: The concern was when result and argument are of different types it caused compilation error. I removed generics to make it more clear.

Comment (fineg74) [reply SYCL/ESIMD/regression/abs_fix_test.cpp]: We are not extremely efficient there as we do the same thing in both cases and I believe we could do better with actual implementation of abs and couple of other functions but I believe it is beyond the scope of this issue.

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**